source/slang/slang-type-system-shared.h

```diff
@@ -58,6 +58,20 @@ FOREACH_BASE_TYPE(DEFINE_BASE_TYPE)
     const int kStdlibTextureIsShadowParameterIndex = 6;
     const int kStdlibTextureIsCombinedParameterIndex = 7;
     const int kStdlibTextureFormatParameterIndex = 8;
+
+    enum class AddressSpace : uint64_t
```

Comment:
My suggestion is similar to Kai's but slightly different in goal.

Currently if I want to add something like a "ThreadLocal" address space pointer I would either:
1. for Metal assign to the Ptr the address space of `ThreadLocal`.
2. for SPIRV assign to the Ptr the address space of `SpvStorageClassFunction`

My suggestion is to make these the semantics the same or for a comment to be added "address space XYZ with Metal maps to address space ABC for SPIRV". This would make it much easier to maintain address-space using code compatibility with SPIRV/Metal.